### PR TITLE
Feature/email notifications

### PR DESCRIPTION
**Pull request by ivanmrsulja**
## Description

Added support for sending notifications to user's email based on their newsletter frequency preference.

## How should this be tested?

A description of what steps someone could take to:

This should be tested in conjunction with [frontend PR](https://github.com/sci2zero/TeslaRIS-frontend/pull/26). Go to profile page and select frequency of newsletter. You should recieve your notifications email based on the CRON configuratim from application.properties.

## Checklist

-   [x] I have tested these changes locally
-   [x] I have added tests that cover these changes (if applicable)
-   [x] I have updated the documentation (if applicable)
-   [x] I have added or updated the necessary comments in the code

### CODE REVIEW

File: src/main/java/rs/teslaris/core/service/impl/commontypes/NotificationServiceImpl.java [hunk 1]
```diff
@@ -26,12 +35,21 @@
...
     private final NewOtherNameNotificationHandler newOtherNameNotificationHandler;
 
     private final AddedToPublicationNotificationHandler addedToPublicationNotificationHandler;
 
+    private final UserAccountIndexRepository userAccountIndexRepository;
+
+    private final EmailUtil emailUtil;
+
+    @Value("${client.address}")
+    private String clientAppAddress;
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/service/impl/commontypes/NotificationServiceImpl.java:51]: This is address of front-end application, right? The name of the property is not intuitive for me.

**Review by chenejac — CHANGES_REQUESTED** [review]
@ivanmrsulja well done. I have posted just one comment which is no crucial. Please think a little bit about name of the property clientAppAddress.

File: src/main/java/rs/teslaris/core/service/impl/commontypes/NotificationServiceImpl.java (hunk 1 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/service/impl/commontypes/NotificationServiceImpl.java:51]: Yes it is, should I change it to `frontend.application.address`?

Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/service/impl/commontypes/NotificationServiceImpl.java:51]: Renamed! :D

**Pull request merged**